packages/block-editor/src/hooks/use-editor-wrapper-styles.native.js

```diff
@@ -0,0 +1,250 @@
+ * @param {string}  blockName            The name of the block.
+ * @param {number}  blockWidth           The width of the block.
+ * @param {boolean} hasParents           Whether the block has parents.
+ * @param {number}  marginHorizontal     Default horizontal margin.
+ * @param {string}  parentBlockAlignment The alignment of the parent block.
+ * @param {string}  parentBlockName      The name of the parent block.
+ * @param {number}  parentWidth          The width of the parent block.
+ * @param {number}  width                The width of the screen.
+ * @return {number} The calculated block margin.
+ */
+export function getBlockMargin(
+	align,
+	blockName,
+	blockWidth,
+	hasParents,
+	marginHorizontal,
+	parentBlockAlignment,
+	parentBlockName,
+	parentWidth,
+	width
+) {
+	const { isContainerRelated, isWider, isWideWidth, isFullWidth } =
+		alignmentHelpers;
+
+	if ( isFullWidth( align ) ) {
+		if ( ! hasParents ) {
+			return 0;
+		}
+		return marginHorizontal;
+	}
+
+	if ( isWideWidth( align ) ) {
+		return marginHorizontal;
+	}
+
+	if (
+		isFullWidth( parentBlockAlignment ) &&
+		! isWider( blockWidth, 'medium' )
+	) {
+		if ( isContainerRelated( blockName ) || isWider( width, 'mobile' ) ) {
+			return marginHorizontal;
+		}
+		return marginHorizontal * 2;
+	}
+
+	if (
+		isContainerRelated( parentBlockName ) &&
+		! isContainerRelated( blockName )
+	) {
+		const isScreenWidthEqual = parentWidth === width;
+		if ( isScreenWidthEqual || isWider( width, 'mobile' ) ) {
+			return marginHorizontal;
+		}
+	}
+
+	return marginHorizontal;
+}
+
+/**
+ * Custom hook to get the styles and margin for the editor wrapper.
+ *
+ * @param {Object}  [props]                    The props for the hook.
+ * @param {string}  props.align                The alignment value.
+ * @param {string}  props.blockName            The name of the block.
+ * @param {number}  props.blockWidth           The width of the block.
+ * @param {string}  props.contentResizeMode    The content resize mode.
+ * @param {boolean} props.hasParents           Whether the block has parents.
+ * @param {number}  props.marginHorizontal     Default horizontal margin.
+ * @param {string}  props.parentBlockAlignment The alignment of the parent block.
+ * @param {string}  props.parentBlockName      The name of the parent block.
+ * @param {number}  props.parentWidth          The width of the parent block.
+ * @param {boolean} [props.reversed=false]     Whether the flex direction should be reversed.
+ * @return {[Array, number]} An array containing the wrapper styles and block margin.
+ */
+export function useEditorWrapperStyles( {
+	align,
+	blockName,
+	blockWidth,
+	contentResizeMode,
+	hasParents,
+	marginHorizontal,
+	parentBlockAlignment,
+	parentBlockName,
+	parentWidth,
+	reversed = false,
+} = {} ) {
+	const { width, height } = useWindowDimensions();
+	const isLandscape = width >= height;
+
+	const blockMargin = useMemo(
+		() =>
+			getBlockMargin(
```

Comment:
In addition to the internal usage of `getBlockMargin` here, I note it is exported as well. Is the intent that this is a public method as well? If not, we should likely avoid exporting it. 

This thought also applies to other functions in this file.

Reply:
> If not, we should likely avoid exporting it.

Related to this point, I'm less inclined to write and maintain automated tests for any "private" methods. Testing the "public" `useEditorWrapperStyles` should be sufficient IMO.